src/od/mockstream.rs

```diff
@@ -0,0 +1,102 @@
+// https://github.com/lazy-bitfield/rust-mockstream/pull/2
+
+use std::io::{Cursor, Read, Result, Error, ErrorKind};
+use std::error::Error as errorError;
+
+/// `FailingMockStream` mocks a stream which will fail upon read or write
+///
+/// # Examples
+///
+/// ```
+/// use std::io::{Cursor, Read};
+///
+/// struct CountIo {}
+///
+/// impl CountIo {
+///     fn read_data(&self, r: &mut Read) -> usize {
+///         let mut count: usize = 0;
+///         let mut retries = 3;
+///
+///         loop {
+///             let mut buffer = [0; 5];
+///             match r.read(&mut buffer) {
+///                 Err(_) => {
+///                     if retries == 0 { break; }
+///                     retries -= 1;
+///                 },
+///                 Ok(0) => break,
+///                 Ok(n) => count += n,
+///             }
+///         }
+///         count
+///     }
+/// }
+///
+/// #[test]
+/// fn test_io_retries() {
+///     let mut c = Cursor::new(&b"1234"[..])
+///             .chain(FailingMockStream::new(ErrorKind::Other, "Failing", 3))
+///             .chain(Cursor::new(&b"5678"[..]));
+///
+///     let sut = CountIo {};
+///     // this will fail unless read_data performs at least 3 retries on I/O errors
+///     assert_eq!(8, sut.read_data(&mut c));
+/// }
+/// ```
+#[derive(Clone)]
+pub struct FailingMockStream {
+    kind: ErrorKind,
+    message: &'static str,
+    repeat_count: i32,
+}
+
+impl FailingMockStream {
+    /// Creates a FailingMockStream
+    ///
+    /// When `read` or `write` is called, it will return an error `repeat_count` times.
+    /// `kind` and `message` can be specified to define the exact error.
+    pub fn new(kind: ErrorKind, message: &'static str, repeat_count: i32) -> FailingMockStream {
+        FailingMockStream { kind: kind, message: message, repeat_count: repeat_count, }
+    }
+
+    fn error(&mut self) -> Result<usize> {
+        if self.repeat_count == 0 {
+            return Ok(0)
+        }
+        else {
```

Comment:
Please put the `else` on the same line as the `}`.  Make changes throughout the code.